apps/prairielearn/src/sync/course-db.ts

```diff
@@ -404,15 +404,30 @@ export async function loadFullCourse(
   perf.end('loadQuestions');
   const courseInstanceInfos = await loadCourseInstances(courseDir);
   const courseInstances: Record<string, CourseInstanceData> = {};
-  for (const courseInstanceId in courseInstanceInfos) {
-    // TODO: is it really necessary to do all the crazy error checking on `lstat` for the assessments dir?
-    // If so, duplicate all that here
-    const assessments = await loadAssessments(courseDir, courseInstanceId, questions);
-    const courseInstance = {
+  for (const [courseInstanceId, courseInstance] of Object.entries(courseInstanceInfos)) {
+    const courseInstanceAccessibleNowOrInFuture = (courseInstance.data?.allowAccess ?? []).some(
```

Comment:
We're technically already computing this deep in `loadCourseInstances`, but I couldn't find a nice way to plumb that back up to this point, so I'm recomputing it 😬